pkg/agent/nodeportlocal/k8s/annotations.go

```diff
@@ -0,0 +1,146 @@
+// +build !windows
+
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/vmware-tanzu/antrea/pkg/util/env"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+)
+
+const NPLAnnotationStr = "nodeportlocal.antrea.io"
+
+// NPLAnnotation is the structure used for setting NodePortLocal annotation on the Pods
+type NPLAnnotation struct {
+	PodPort  int    `json:"podPort"`
+	NodeIP   string `json:"nodeIP"`
+	NodePort int    `json:"nodePort"`
+}
+
+func toJSON(serialize interface{}) string {
+	jsonMarshalled, _ := json.Marshal(serialize)
+	return string(jsonMarshalled)
+}
+
+func isNodePortInAnnotation(s []NPLAnnotation, nodeport int) bool {
+	for _, i := range s {
+		if i.NodePort == nodeport {
+			return true
+		}
+	}
+	return false
+}
+
+func assignPodAnnotation(pod *corev1.Pod, nodeIP string, containerPort, nodePort int) {
+	var err error
+	current := pod.Annotations
+	if current == nil {
+		current = make(map[string]string)
+	}
+
+	klog.V(2).Infof("Building annotation for Pod: %s\tport: %v --> %v:%v", pod.Name, containerPort, nodeIP, nodePort)
+
+	var annotations []NPLAnnotation
+	if current[NPLAnnotationStr] != "" {
+		if err = json.Unmarshal([]byte(current[NPLAnnotationStr]), &annotations); err != nil {
+			klog.Warningf("Unable to unmarshal NPLEP annotation: %v", current[NPLAnnotationStr])
```

Comment:
what is "NPLEP"? Suggest using "NodePortLocal Endpoint annotation" in user-facing log messages

other occurrences below